### PR TITLE
Components

### PR DESCRIPTION
**Pull request by george-swift**
### Milestone 2
In accordance with project specifications:
- A components folder was created with four components `App, Display, Button Panel and Button`
- `App` is stateless for now hence it is a function component. It renders the `Display` and `Button Panel` components
- The `Display` component accepts `result` prop. The prop type is defined and the default prop is `0`.
- The `Button Panel` component renders the required five groups of buttons, each with a `name` prop.
- The `Button` component renders a single button. It requires a `name` prop of type string.

### Additional Information
- All components and functions used in this milestone have been tested. This is the reason for creating the `__tests__` directory
- For styling purposes, the `Button` component uses the `name` prop to define other attributes
- The data folder contains a JSON file that lists the button groups and a module for button keys. These are used in the `Button Panel` and `Button` components respectively to create components dynamically.
- In the `Display` component, after meeting general requirements, I wish to show problems above the result below, hence the `div className='problem'`

### CODE REVIEW

File: src/components/Display.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    </div>
+  </div>
+);
+
+Display.propTypes = {
+  result: PropTypes.number,
+};
+
+Display.defaultProps = {
+  result: 0,
```
Comment (afanimartin) [inline src/components/Display.js:17]: - This should be a string: `result: '0'`

File: src/components/Display.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  <div className="col-12 h-25 text-white">
+    <div className="problem" />
+    <div className="result">
+      {result}
+    </div>
+  </div>
+);
+
+Display.propTypes = {
+  result: PropTypes.number,
```
Comment (afanimartin) [inline src/components/Display.js:13]: - Validate a string value: `result: PropTypes.string`

**Review by afanimartin — CHANGES_REQUESTED** [review]
## Changes Required 🙏🏿 🙏🏿 🙏🏿 
Nice work on this milestone. Your codebase is well structured and most requirements have been satisfied.
### Highlights 👍🏿 👍🏿 👍🏿 
- Correct implementation of `App`, `Button`, and `ButtonPanel` components.

### Change(s) 🔕 🔕 🔕 
- The `result` object should be of a type string not a number. Check comments.

### Martin Afani
#### Slack: whiz25

**Review by Berabjesus — APPROVED** [review]
# **Status: PROJECT APPROVED** 
Hello @george-swift great job implementing the changes required
**All the required tasks have been fulfilled in this project. I will go ahead and mark this as complete. 👍** 

I wish you all the very best in your next tasks:1st_place_medal:
#### **If you have any questions, let me know by leaving a comment here and mentioning me. I will be happy to help** 
**HAPPY CODING :computer:[Bereket Beshane](https://bereketababu.me)** | [LinkedIn](https://linkedin.com/in/bereketbeshane) | [GitHub](https://github.com/Berabjesus)

**Pull request merged**